core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -715,17 +718,25 @@ class HttpGoogleServicesDAO(
   }
 
   def diagnosticBucketRead(userInfo: UserInfo, bucketName: String): Future[Option[ErrorReport]] = {
-    implicit val service = GoogleInstrumentedService.Storage
-    Future {
-      val getter = getStorage(getUserCredential(userInfo)).buckets().get(bucketName)
-      try {
-        blocking {
-          executeGoogleRequest(getter)
-        }
-        None
-      } catch {
-        case t: HttpResponseException => Some(ErrorReport(StatusCode.int2StatusCode(t.getStatusCode), t.getMessage))
+    // we make requests to the target bucket as the default pet, if the user only has read access.
+    // the default pet is created in a project without APIs enabled. Due to Google issue #16062674, we cannot
+    // call the GCS JSON API as the default pet; we must use the XML API. In turn, this means we cannot use
+    // the GCS client library (which internally calls the JSON API); we must hand-code a call to the XML API.
+
+    // the "proper" request to make into the XML API is HEAD-bucket; see https://cloud.google.com/storage/docs/xml-api/overview.
+    // however, the akka-http client is vulnerable to connection-pool exhaustion with HEAD requests; see akka/akka-http#1495.
+    // therefore, we make a request to GET /?storageClass. Since all we care about is the 200 vs. 40x status code
+    // in the response, this is an equivalent request.
+    val bucketUrl = s"https://$bucketName.storage.googleapis.com/?storageClass"
+    val bucketRequest = httpClientUtils.addHeader(RequestBuilding.Head(bucketUrl), Authorization(userInfo.accessToken))
+
+    httpClientUtils.executeRequest(http, bucketRequest) map { httpResponse =>
+      httpResponse.status match {
+        case StatusCodes.OK => None
+        case x => Some(ErrorReport(x, x.defaultMessage()))
```

Comment:
instead of x. defaultMessage I would use the httpResponse body

Reply:
In this case, the httpResponse body is going to be an XML error message, like:
```
<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message></Error>
```
Which I'd rather not return verbatim, and I'd also like to avoid parsing.